tests/integration/suite/actors/rebalancing-double-activation/rebalancing-double-activation.go

```diff
@@ -0,0 +1,416 @@
+/*
+Copyright 2023 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or impliei.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package healthz
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	chi "github.com/go-chi/chi/v5"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc"
+	grpcinsecure "google.golang.org/grpc/credentials/insecure"
+
+	placementv1pb "github.com/dapr/dapr/pkg/proto/placement/v1"
+	"github.com/dapr/dapr/tests/integration/framework"
+	"github.com/dapr/dapr/tests/integration/framework/process/daprd"
+	prochttp "github.com/dapr/dapr/tests/integration/framework/process/http"
+	"github.com/dapr/dapr/tests/integration/framework/process/placement"
+	"github.com/dapr/dapr/tests/integration/framework/util"
+	"github.com/dapr/dapr/tests/integration/suite"
+)
+
+func init() {
+	suite.Register(new(rebalancingDoubleActivation))
+}
+
+// Number of iterations for the test
+const iterations = 30
+
+// rebalancingDoubleActivation tests that during rebalancing, reminders do not cause actors to be activated on 2 separate hosts.
+type rebalancingDoubleActivation struct {
+	daprd              [2]*daprd.Daprd
+	srv                [2]*prochttp.HTTP
+	handler            [2]*httpServer
+	place              *placement.Placement
+	placementStream    placementv1pb.Placement_ReportDaprStatusClient
+	activeActors       []atomic.Bool
+	doubleActivationCh chan string
+}
+
+func (i *rebalancingDoubleActivation) Setup(t *testing.T) []framework.Option {
+	i.activeActors = make([]atomic.Bool, iterations)
+	i.doubleActivationCh = make(chan string)
+
+	// Remove the database if present
+	// Ignore errors here
+	_ = os.Remove("tmp/test-data.db")
+
+	// Init placement
+	i.place = placement.New(t)
+
+	// Init two instances of daprd, each with its own server
+	for j := 0; j < 2; j++ {
+		i.handler[j] = &httpServer{
+			activeActors:       i.activeActors,
+			doubleActivationCh: i.doubleActivationCh,
+		}
+		i.srv[j] = prochttp.New(t, prochttp.WithHandler(i.handler[j].NewHandler(j)))
+		i.daprd[j] = daprd.New(t, daprd.WithResourceFiles(`
+apiVersion: dapr.io/v1alpha1
+kind: Component
+metadata:
+  name: mystore
+spec:
+  type: state.sqlite
+  version: v1
+  metadata:
+    - name: connectionString
+      value: 'tmp/test-data.db'
+    - name: busyTimeout
+      value: '10s'
+    - name: disableWAL
+      value: 'true'
+    - name: actorStateStore
+      value: 'true'
```

Comment:
Why are we using sqlite over in-memory? I _think_ it is because we want to use a state store which is slower? Please add a comment explaining this.

Reply:
I explained it above. There are 2 sidecars here so we need a state store that is shared among them.